lib/runtime/src/lib.rs

```diff
@@ -268,11 +345,31 @@ pub fn compiler_for_backend(backend: Backend) -> Option<Box<dyn Compiler>> {
             #[cfg(feature = "default-backend-llvm")]
             return Some(Box::new(wasmer_llvm_backend::LLVMCompiler::new()));
         }
-
-        #[cfg(not(all(feature = "llvm", feature = "singlepass", feature = "cranelift")))]
-        _ => None,
     }
 }
 
 /// The current version of this crate.
 pub const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use std::str::FromStr;
+
+    #[test]
+    fn str_repr_matches() {
+        // if this test breaks, think hard about why it's breaking
+        // can we avoid having these be different?
+
+        for &backend in &[
+            #[cfg(feature = "cranelift")]
+            Backend::Cranelift,
+            #[cfg(feature = "llvm")]
+            Backend::LLVM,
+            #[cfg(feature = "singlepass")]
+            Backend::Singlepass,
```

Comment:
We should add `Auto` here as well

Reply:
I skipped `Auto` here in propose since auto is not a real backend (but more like a feature that a frontend will use/define)